### PR TITLE
Add Rico Dataset

### PR DESCRIPTION
**Pull request by ncoop57**
Hi there!

I'm wanting to add the Rico datasets for software engineering type data to y'alls awesome library. However, as I have started coding, I've ran into a few hiccups so I thought it best to open the PR early to get a bit of discussion on how the Rico datasets should be added to the `datasets` lib.

1) There are 7 different datasets under Rico and so I was wondering, should I make a folder for each or should I put them as different configurations of a single dataset?
You can see the datasets available for Rico here: http://interactionmining.org/rico

2) As of right now, I have a semi working version of the first dataset which has pairs of screenshots and hierarchies from android applications. However, these screenshots are very large (1440, 2560, 3) and there are 66,000 of them so I am not able to perform the processing that  the `datasets` lib does after downloading and extracting the dataset since I run out of memory very fast. Is there a way to have `datasets` lib not put everything into memory while it is processing the dataset?

2.1) If there is not a way, would it be better to just return the path to the screenshots instead of the actual image?

3) The hierarchies are JSON objects and looking through the documentation of `datasets`, I didn't see any feature that I could use for this type of data. So, for now I just have it being read in as a string, is this okay or should I be doing it differently?

4) One of the Rico datasets is a bunch of animations (GIFs), is there a `datasets` feature that I can put this type of data into or should I just return the path as a string?

I appreciate any and all help I can get for this PR, I think the Rico datasets will be an awesome addition to the library :nerd_face: !

### CODE REVIEW

File: datasets/rico/rico.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        #                 "sentence": data["sentence"],
+        #                 "option1": data["option1"],
+        #                 "answer": "" if split == "test" else data["answer"],
+        #             }
+        #         else:
+        #             yield id_, {
+        #                 "sentence": data["sentence"],
+        #                 "option2": data["option2"],
+        #                 "second_domain_answer": "" if split == "test" else data["second_domain_answer"],
+        #             }
```
[the remark this answers is not in the document]
Comment (ncoop57) [reply datasets/rico/rico.py:283]: Yeah, just haven't had a chance yet to add these. Also, one of them is like 256 GBs compressed so gonna have to get creative to work with that one lol

File: datasets/code_clippy/code_clippy.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        
+        #TODO(high priority): changes this to load from the-eye
+        # data_dir = dl_manager.download_and_extract(_URLs)
+        data_dir = self.config.data_dir
```
[the remark this answers is not in the document]
Comment (ncoop57) [reply datasets/code_clippy/code_clippy.py:108]: Yes it is, it is located here: https://the-eye.eu/public/AI/training_data/code_clippy_data/

File: datasets/code_clippy/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  - [Licensing Information](#licensing-information)
+  - [Citation Information](#citation-information)
+  - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** https://the-eye.eu/public/AI/training_data/code_clippy_data/
+- **Repository:** https://github.com/ncoop57/gpt-code-clippy
+- **Paper:** [Not yet :)]
+- **Leaderboard:** [Not yet :)]
```
[the remark this answers is not in the document]
Comment (ncoop57) [reply datasets/code_clippy/README.md:38]: Me too :nerd_face:  also sorry this was not supposed to be part of the rico dataset PR. Is there any easy way that you know of that I can separate the two? This code clippy is already part of a different PR that my team has been working on, but my envs got combined by mistake

[the remark this answers is not in the document]
Comment (lhoestq) [reply datasets/code_clippy/README.md:38]: You can just remove those files from this PR I guess, and then you or your team can open a separate PR for code clippy

File: datasets/rico/rico.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhoestq) [reply datasets/rico/rico.py:283]: Let me know if we can help with this :)

**Pull request closed without merging**